src/sql/src/parsers/copy_parser.rs

```diff
@@ -126,6 +134,40 @@ impl ParserContext<'_> {
         }
     }
 
+    fn parse_copy_query_to(&mut self) -> Result<CopyQueryTo> {
+        self.parser
+            .expect_token(&Token::LParen)
+            .with_context(|_| error::UnexpectedSnafu {
+                expected: "'('",
+                actual: self.peek_token_as_string(),
+            })?;
+        let query = self.parse_query()?;
+        self.parser
+            .expect_token(&Token::RParen)
+            .with_context(|_| error::UnexpectedSnafu {
+                expected: "')'",
+                actual: self.peek_token_as_string(),
+            })?;
+        self.parser
+            .expect_keyword(Keyword::TO)
+            .context(error::SyntaxSnafu)?;
+        let (with, connection, location, limit) = self.parse_copy_parameters()?;
+        if limit.is_some() {
+            return error::InvalidSqlSnafu {
+                msg: "limit is not supported",
+            }
+            .fail();
+        }
```

Comment:
Is it possible to execute an SQL command like this? 
```sql
COPY (select host, cpu, ts from demo where host = 'host2' limit 1) TO '${SQLNESS_HOME}/export/demo.csv' WITH (format='csv');
```

Reply:
yep, if it requires limit, then it should not be in `copy to` but in `query`